theatre/studio/src/StudioStore/createTransactionPrivateApi.ts

```diff
@@ -69,179 +69,210 @@ function forEachDeepSimplePropOfCompoundProp(
+    const setStaticOrKeyframeProp = <T>(
+      value: T,
+      propConfig: PropTypeConfig_AllSimples,
+      path: PathToProp,
+    ) => {
+      if (value === undefined || value === null) {
+        return
+      }
 
-        //   propConfig.validate(_value)
-        // }
+      const deserialized = cloneDeepSerializableAndPrune(
+        propConfig.deserializeAndSanitize(value),
+      )
+      if (deserialized === undefined) {
+        throw new Error(
+          `Invalid value ${userReadableTypeOfValue(
+            value,
+          )} for object.props${path
+            .map((key) => `[${JSON.stringify(key)}]`)
+            .join('')} is invalid`,
+        )
+      }
 
-        const setStaticOrKeyframeProp = <T>(
-          value: T,
-          propConfig: PropTypeConfig_AllSimples,
-          path: PathToProp,
-        ) => {
-          if (value === undefined || value === null) {
-            return
-          }
+      const propAddress = {...root.address, pathToProp: path}
+
+      const trackId = get(sequenceTracksTree, path) as $FixMe as
+        | SequenceTrackId
+        | undefined
+
+      if (typeof trackId === 'string') {
+        const seq = root.sheet.getSequence()
+        seq.position = seq.closestGridPosition(seq.position)
+        stateEditors.coreByProject.historic.sheetsById.sequence.setKeyframeAtPosition(
+          {
+            ...propAddress,
+            trackId,
+            position: seq.position,
+            value: value as $FixMe,
+            snappingFunction: seq.closestGridPosition,
+          },
+        )
+      } else {
+        stateEditors.coreByProject.historic.sheetsById.staticOverrides.byObject.setValueOfPrimitiveProp(
+          {...propAddress, value: value as $FixMe},
+        )
+      }
+    }
+
+    if (propConfig.type === 'compound') {
+      const pathToTopPointer = options.pathToProp
+
+      const lengthOfTopPointer = pathToTopPointer.length
+      // If we are dealing with a compound prop, we recurse through its
+      // nested properties.
+      forEachDeepSimplePropOfCompoundProp(
+        propConfig,
+        pathToTopPointer,
+        (primitivePropConfig, pathToProp) => {
+          const pathToPropInProvidedValue = pathToProp.slice(lengthOfTopPointer)
 
-          const deserialized = cloneDeepSerializableAndPrune(
-            propConfig.deserializeAndSanitize(value),
-          )
-          if (deserialized === undefined) {
+          const v = getDeep(_value, pathToPropInProvidedValue)
+          if (typeof v !== 'undefined') {
+            setStaticOrKeyframeProp(v, primitivePropConfig, pathToProp)
+          } else {
             throw new Error(
-              `Invalid value ${userReadableTypeOfValue(
-                value,
-              )} for object.props${path
+              `Property object.props${pathToProp
                 .map((key) => `[${JSON.stringify(key)}]`)
-                .join('')} is invalid`,
+                .join('')} is required but not provided`,
```

Comment:
It's actually invalid. If it is invalid, it'll be ommitted by `cloneDeepSerializeAndPrune()`

Reply:
Happy to change it. Can we say more words to improve this message?